### ISSUE TITLE
`settings[:gemfile]` from `.bundle/config` file is not respected when doing `require 'bundle/setup'`

### ISSUE THREAD

**Issue by faucct** [bundler/bundler#6589]
Gemfile configuration from `.bundle/config` is only respected when you use `Bundler::CLI`. Can someone tell me why, please?

**Comment by segiddins** [comment]
Because Bundler resolves `.bundle/config` relative to the current Gemfile, so it needs to have a Gemfile first before looking up settings. You'll need to use `$BUNDLE_GEMFILE` if you want to control that path

**Comment by faucct** [comment]
> Bundler resolves .bundle/config relative to the current Gemfile

No, it is being resolved relative to `Dir.pwd` and does not need a gemfile: https://github.com/bundler/bundler/blob/master/lib/bundler/shared_helpers.rb#L47
Does anyone mind me sharing this `Bundler::CLI` functionality with `bundler/setup`?

**Comment by deivid-rodriguez** [comment]
Personally, this sounds like something that could be fixed to me.

**Comment by deivid-rodriguez** [comment]
@faucct I forgot to ping you about this, I fixed this in #6713, in case you want to give that branch a try! You'll need to use the env variable `BUNDLE_CONFIG_RELATIVE_TO_CWD` in order to enable the feature.

**Comment by deivid-rodriguez** [comment]
@faucct Did you end up trying my PR?

**Comment by faucct** [comment]
Yes: I didn't have the access to the project with the problem at that time anymore.

**Comment by deivid-rodriguez** [comment]
So, to confirm I understood (the initial "Yes" in your reply confused me), you were NOT able to try this PR and confirm it fixes the problem you had at the time, right?

**Comment by faucct** [comment]
Yes, you’ve got me right.
